### PR TITLE
Remove isTestEnv

### PR DESCRIPTION
**Pull request by Nokel81**
- Make all migration declarations injectable using a token

- Make all migrations injectable using a token for the renderer side
  (instead of clearing them based on the truthiness of ipcRenderer)

Signed-off-by: Sebastian Malton <sebastian@malton.name>

### CODE REVIEW

File: src/main/migrations/hotbar-store/5.0.0-beta.10.injectable.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+      version: "5.0.0-beta.10",
+      run(store) {
+        const rawHotbars = store.get("hotbars");
+        const hotbars: Hotbar[] = Array.isArray(rawHotbars) ? rawHotbars.filter(h => h && typeof h === "object") : [];
+
+        // Hotbars might be empty, if some of the previous migrations weren't run
+        if (hotbars.length === 0) {
+          const hotbar = getEmptyHotbar("default");
+
+          const di = getLegacyGlobalDiForExtensionApi();
```
Comment (jansav) [inline src/main/migrations/hotbar-store/5.0.0-beta.10.injectable.ts]: No need for this anymore :)

File: src/main/migrations/hotbar-store/5.0.0-beta.10.injectable.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+          /**
+           * Finally, make sure that the catalog entity hotbar item is in place.
+           * Just in case something else removed it.
+           *
+           * if every hotbar has elements that all not the `catalog-entity` item
+           */
+          if (hotbars.every(hotbar => hotbar.items.every(item => item?.entity?.uid !== "catalog-entity"))) {
+            // note, we will add a new whole hotbar here called "default" if that was previously removed
+            const di = getLegacyGlobalDiForExtensionApi();
```
Comment (jansav) [inline src/main/migrations/hotbar-store/5.0.0-beta.10.injectable.ts]: No need anymore

File: src/renderer/migrations/hotbar-store.injectable.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import { hotbarStoreMigrationsInjectionToken } from "../../common/hotbars/migrations";
+
+const hotbarStoreMigrationsInjectable = getInjectable({
+  id: "hotbar-store-migrations",
+  instantiate: () => undefined,
+  injectionToken: hotbarStoreMigrationsInjectionToken,
+});
```
Comment (jansav) [inline src/renderer/migrations/hotbar-store.injectable.ts:12]: For the future: Requirement for these just highlights the fact that running migrations should be extracted from the stores in a way that `migrations` is concept of main. 

That would mean that we could have `runMigrations` runnable which is ran at `onLoadOfApplication`.

File: src/common/base-store.ts [hunk 4]
```diff
@@ -36,7 +40,7 @@ export abstract class BaseStore<T> extends Singleton {
 
   readonly displayName: string = this.constructor.name;
 
-  protected constructor(protected params: BaseStoreParams<T>) {
+  protected constructor(protected readonly depenendices: BaseStoreDependencies, protected params: BaseStoreParams<T>) {
```
Comment (jansav) [inline src/common/base-store.ts]: typo `depenendices` -> `dependencies`

**Comment by jansav** [comment]
> Pipe dependencies through to BaseStore without using getLegacyDi

This is big thing. It will allow us to do a lot. Love it.

File: src/behaviours/cluster/__snapshots__/order-of-sidebar-items.test.tsx.snap [hunk 5]
```diff
@@ -359,7 +359,10 @@ exports[`cluster - order of sidebar items when rendered renders 1`] = `
               >
```
Comment (jansav) [inline src/behaviours/cluster/__snapshots__/order-of-sidebar-items.test.tsx.snap]: I think these are left-overs from master. I'll verify and create separate PR for these if so.

**Review by jansav — COMMENTED** [review]
Man, you are doing good stuff here. :) 

There's still some improvements to be done in the future as I mentioned in other comments. 

For now, some small tweaks here and there.

File: src/behaviours/cluster/__snapshots__/order-of-sidebar-items.test.tsx.snap (hunk 5 above)
Comment (jansav) [reply src/behaviours/cluster/__snapshots__/order-of-sidebar-items.test.tsx.snap]: https://github.com/lensapp/lens/pull/5815

**Comment by Nokel81** [comment]
Closing as this has too many conflicts with the PRs to fix release details

**Pull request closed without merging**